src/openforms/registrations/contrib/microsoft_graph/plugin.py

```diff
@@ -12,18 +14,42 @@
 )
 from openforms.submissions.models import Submission, SubmissionReport
 from openforms.submissions.tasks.registration import set_submission_reference
+from openforms.template import render_from_string
 
 from ...base import BasePlugin
 from ...exceptions import NoSubmissionReference, RegistrationFailed
 from ...registry import register
+from .config import MicrosoftGraphOptionsSerializer
 
 
 @register("microsoft-graph")
 class MSGraphRegistration(BasePlugin):
     verbose_name = _("Microsoft Graph (OneDrive/SharePoint)")
+    configuration_options = MicrosoftGraphOptionsSerializer
+
+    def _get_folder_name(self, submission: Submission, options: dict) -> "Path":
+        date = timezone.now().date()
+        folder_path = render_from_string(
+            options["folder_path"],
+            {
+                "year": "{date:%Y}".format(date=date),
+                "month": "{date:%m}".format(date=date),
+                "day": "{date:%d}".format(date=date),
+            },
+        )
+
+        return (
+            Path(folder_path)
```

Comment:
this should also be `PurePosixPath` to enforce the correct slashes on different environments.

Additionally, you don't need to wrap the second/third segment in the `Path` constructor:

```py
return PurePosixPath(
    folder_path,
    slugify(submission.form.admin_name,
    submission.public_registration_reference,
)
```
should work and have the same result.